flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTask.java

```diff
@@ -143,7 +143,8 @@ private CheckpointedInputGate createCheckpointedInputGate() {
                 inputGates,
                 getEnvironment().getMetricGroup().getIOMetricGroup(),
                 getTaskNameWithSubtaskAndId(),
-                mainMailboxExecutor);
+                mainMailboxExecutor,
+                timerService);
```

Comment:
One more concern: this `timerService` is used for user timers; their execution and even maintanance (there can be unlimited number of such timers) may potentially affect (delay) checkpointing.

WDYT? Should we use a dedicated timer/Service?

Reply:
It's cerntainly not a general case but I'm wouldn't be sure that it can't happen in some setups.
And if it's easy to fix and doesn't add overhead then why not to do it?